src/client/bridge/gateway/shard_runner.rs

```diff
@@ -229,6 +237,23 @@ impl ShardRunner {
 
             retain(&mut self.reaction_filters, |f| f.send_reaction(&reaction));
         }
+
+        // Avoid the clone if there is no interaction filter.
+        #[cfg(all(feature = "unstable_discord_api", feature = "collector"))]
+        if !self.component_interaction_filters.is_empty() {
+            let interaction = Arc::new(match &event {
+                Event::InteractionCreate(ref interaction_event) => {
+                    if interaction_event.interaction.kind == InteractionType::MessageComponent {
+                        Arc::new(interaction_event.interaction.clone())
+                    } else {
+                        return;
+                    }
+                },
+                _ => return,
```

Comment:
In case more filters are added in the future, it shouldn't return on other events here or it will skip later filters like in #1374.  Maybe replace with an `if let Event::InteractionCreate(ref interaction_event) = &event` instead of a match statement.